packages/core/contracts/optimistic-asserter/implementation/OptimisticAsserter.sol

```diff
@@ -100,8 +95,8 @@ contract OptimisticAsserter is OptimisticAsserterInterface, Lockable, Ownable {
 
         require(assertions[assertionId].asserter == address(0), "Assertion already exists");
         // TODO [GAS] caching identifier whitelist and collateral currency whitelist
-        require(_getIdentifierWhitelist().isIdentifierSupported(identifier), "Unsupported identifier");
-        require(_getCollateralWhitelist().isOnWhitelist(address(currency)), "Unsupported currency");
+        require(cachedUmaParams.supportedIdentifiers[identifier], "Unsupported identifier");
```

Comment:
there are some implications with this design that we should consider. the way you've structured it here _requires_ that you first call `syncUmaParams` _before_ you make an assertion for a new currency or a new identifier. this is a poor UX. Rather, we should construct these requires to _first_ try find values in the cache, and if they cant be found, then try and fetch them from the UMA identifier whitelist and collateral whitelist. in doing it this way we effectively _auto populate the cache_ on the first use of the caching logic and subsequently use values from the cache.

you might need to define some internal functions to do this (such as `isOnWhiteList`) that do this sequential logic.

wdyt?

Reply:
Agree, this totally makes sense. Will add this.